### PR TITLE
added sklearn precision score metric and the test for it

### PR DESCRIPTION
**Pull request by EngineerLambda**
# PR Description
I added a metric to the sklearn frontend.
Under the classification metrics, I added precision score

## Related Issue
No related issue, I just checked through the frontend codes and found that it was missing that

## Checklist

- [x] Did you add a function? Yes
- [x] Did you add the tests? Yes
- [x] Did you run your tests and are your tests passing? Some passed
- [X ] Did pre-commit not fail on any check? No
- [x] Did you follow the steps we provided? Yes

### Socials

twitter.com/abdulsomad_me

### CODE REVIEW

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@EngineerLambda Looks good, just left a few comments. Also the test seems to be failing, please could you look into this? Thanks!

File: ivy/functional/frontends/sklearn/metrics/_classification.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import ivy
+import numpy as np  # To handle the nans
```
[the remark this answers is not in the document]
Comment (EngineerLambda) [reply ivy/functional/frontends/sklearn/metrics/_classification.py:2]: Old habits, I actual made the pull request as it was required for an internship role at ivy that's already closed now. Had to do this in a short time, as I was dealing with school stuffs, personal projects and new learnings.

File: ivy/functional/frontends/sklearn/metrics/_classification.py [hunk 2]
```diff
@@ -17,3 +19,103 @@ def accuracy_score(y_true, y_pred, *, normalize=True, sample_weight=None):
...
+    - y_pred (ivy array): Predicted target values.
+    - labels (ivy array or None): List of labels to include (optional).
+    - pos_label (int or str): The positive class label (default is 1).
+    - average (str): Determines how precision is averaged.
+    - sample_weight (ivy array or None): Sample weights (optional).
+    - zero_division (str or float): Sets the value to return when there is a zero division.
+
+    Returns:
+    - Precision (array of floats or float): The calculated precision score(s).
+    """
```
[the remark this answers is not in the document]
Comment (EngineerLambda) [reply ivy/functional/frontends/sklearn/metrics/_classification.py:50]: Thanks, Lol

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**